runtime/src/bank/bank_hash_details.rs

```diff
@@ -119,7 +116,8 @@ pub struct SlotDetails {
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, Default)]
```

Comment:
I don't understand. where/when do we serde this?

Reply:
For debug purposes. Two places:
- ReplayStage when the node realizes it has diverged and dumps a view of current state (when were were hunting non-deterministic bugs)
- ledger-tool, to create the same file so we can diff them between validators/versions/etc

Reply:
since we're changing the format, will this mean the differ will fail during the transition if we log a file from 2 different versions? Of course, a ledger tool from the same version will create an equivalent file.

Reply:
i am a little concerned with someone wasting time if the file formats are different because you'll get subtley garbage data (worst case) or best case some type of eof or extra byte error.
This is probably fine - just me being concerned about serde issues.

Reply:
Agreed with Brooks - it will be a one line diff in the hypothetical scenario where someone is debugging v2.1 vs v2.2. The output file is a pretty printed json file, so it will quite obvious that one field is no longer present vs something like a hash differing and wondering what the heck caused it